src/main/java/com/arcbees/analytics/server/options/ServerOptionsCallback.java

```diff
@@ -68,17 +68,17 @@ public void onCallback(String options) {
     }
 
     @Override
-    public void putBoolean(String fieldName, boolean value) {
+    public synchronized void putBoolean(String fieldName, boolean value) {
         putText(fieldName, value ? "1" : "0");
     }
 
     @Override
-    public void putNumber(String fieldName, double value) {
+    public synchronized void putNumber(String fieldName, double value) {
         putText(fieldName, value + "");
     }
 
     @Override
-    public void putText(String fieldName, String value) {
+    public synchronized void putText(String fieldName, String value) {
         if (value == null) {
             options.remove(ProtocolTranslator.getFieldName(fieldName));
```

Comment:
If you synchronize access to some object, you need to synchronize **all** accesses to it: writes, and reads (i.e. `getOptions` needs a `synchronized` too).

To avoid contention (I have absolutely no idea how this code is used, so maybe it's a non-issue), you might want to just use a `ConcurrentHashMap` (with no need to add `synchronized` keywords anywhere); or possibly use a `ReentrantReadWriteLock` around a basic `HashMap`, locking the `readLock()` around the `for` loop in `getOptions`.